src/AzureAppConfigurationImpl.ts

```diff
@@ -1,28 +1,46 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { AppConfigurationClient, ConfigurationSetting, ListConfigurationSettingsOptions } from "@azure/app-configuration";
+import { AppConfigurationClient, ConfigurationSetting, ConfigurationSettingId, GetConfigurationSettingOptions, GetConfigurationSettingResponse, ListConfigurationSettingsOptions } from "@azure/app-configuration";
+import { RestError } from "@azure/core-rest-pipeline";
 import { AzureAppConfiguration } from "./AzureAppConfiguration";
 import { AzureAppConfigurationOptions } from "./AzureAppConfigurationOptions";
 import { IKeyValueAdapter } from "./IKeyValueAdapter";
 import { JsonKeyValueAdapter } from "./JsonKeyValueAdapter";
-import { KeyFilter, LabelFilter } from "./types";
+import { DefaultRefreshIntervalInMs, MinimumRefreshIntervalInMs } from "./RefreshOptions";
+import { Disposable } from "./common/disposable";
 import { AzureKeyVaultKeyValueAdapter } from "./keyvault/AzureKeyVaultKeyValueAdapter";
+import { RefreshTimer } from "./refresh/RefreshTimer";
 import { CorrelationContextHeaderName } from "./requestTracing/constants";
 import { createCorrelationContextHeader, requestTracingEnabled } from "./requestTracing/utils";
-import { SettingSelector } from "./types";
+import { KeyFilter, LabelFilter, SettingSelector } from "./types";
 
-export class AzureAppConfigurationImpl extends Map<string, unknown> implements AzureAppConfiguration {
+/**
+ * Key-value identifier, used as key of a Map.
+ * A primitive string type is actually used, because operator === is used to compare identifiers and we cannot override it.
+ */
+type KeyValueIdentifier = string; // key::label
+function toKeyValueIdentifier(key: string, label: string | undefined): KeyValueIdentifier {
+    return `${key}::${label ?? ""}`;

```

Comment:
> This approach might produce same KeyValueIdentifier for different key-label combinations. For eg, these will have the same output:
> 
> key="AppName" and label="Dev"
> key="AppName::Dev" and label=""
> 
> Can we use a different model to uniquely identify key/label combinations?
> -- Avani

Reply:
@avanigupta 
> key="AppName" and label="Dev", id is "AppName::Dev"
> key="AppName::Dev" and label="", id is "AppName::Dev::"

Unproper example, they are not the same. 

Another example, in below cases both have id "A::B::C".
- key="A::B" label="C"
- key="A"  label="B::C"

Reply:
Using such simple formatted string will always have corner cases uncovered.

Trying making it more complicated, e.g. using `key:${key},label:${label}` as id, we might cover most **valid** cases. But we can still construct corner case as below:
`key`="A,label:B", `label`="C", `id`="key:A,label:B,label:C"
`key`="A", `label`="B,label:C", `id`="key:A,label:B,label:C"

Reply:
Googled and found some are using `JSON.stringify()`.  But it doesn't solve the problem.
1. `{key: a, label:b}` and `{label:b, key:a}` are serialized into different strings, which is not expected. But this can be solved by sorting properties by keys, and we don't have nested values here.
2. `{key: a, label:b}` is serialized as a string `{"key":"a","label":"b"}`, but essentially it's also a type of formatted string, with `{"key":"` as prefix, `","label":"` as separator, `"}` as postfix. Counter-example can be easily constructed as above.

Reply:
Why do we have to use strings at all? Is it possible to create a new type with key and label fields, and a custom comparator to define equality? I'm imagining something like our .NET [KeyValueIdentifier](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Models/KeyValueIdentifier.cs)

Reply:
There is currently a [proposal](https://github.com/tc39/proposal-record-tuple#usage-in-mapsetweakmapweakset) to add Records and Tuples (basically immutable Objects and Arrays) to JavaScript.

So I would suggest:
a) either we use a relatively complicated separator for formatted string, e.g. as above, `key:${key},label:${label}`.
b) or we don't use the Map, but an Array, and iterate the array to find the matched element every time (of course more cost)...


WDYT? @avanigupta

Reply:
> and a custom comparator to define equality?

No, it's impossible in JavaScript...(yes, today, in 2023). It forces to use `===` to compare keys, and we cannot override operator in JavaScript.
FYI. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#key_equality

Reply:
> a) either we use a relatively complicated separator for formatted string, e.g. as above, `key:${key},label:${label}`. 

I am hesitant to use this approach because it will always leave room for some error.

> b) or we don't use the Map, but an Array, and iterate the array to find the matched element every time (of course more cost)...

This doesn't sound too bad. From the current code, it looks like we're not really searching in the map anyway. We're mostly using the map iteratively. Moreover, since the map only holds watched settings, and we know that users only provide a few sentinel keys (it's difficult to setup 100s of watched settings) - iterating the array should be fine too.

Reply:
Agree. Let’s use array. We should ensure the correctness first, and improve the performance later if necessary.

Reply:
fixed in d0440dc, changing sentinels to array